backend/nest-app/src/users/users.repository.ts

```diff
@@ -0,0 +1,42 @@
+import { MikroORM } from '@mikro-orm/core';
+import { User } from './users.entity';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { NotFoundException } from '@nestjs/common';
+
+export class UserRepository {
+  constructor(private readonly orm: MikroORM) {}
+
+  async findAllUsers(): Promise<User[]> {
+    const userRepository = this.orm.em.getRepository(User);
+    return userRepository.findAll();
+  }
+
+  async findOneById(id: number): Promise<User | null> {
+    const userRepository = this.orm.em.getRepository(User);
+    const user = await userRepository.findOne({ id });
+    if (!user) {
+      throw new NotFoundException(`User with ID ${id} not found`);
+    }
+    return user;
+  }
+
+  async findUserByEmail(email: string): Promise<User | null> {
+    const userRepository = this.orm.em.getRepository(User);
+    const user = await userRepository.findOne({ email });
+    if (!user) {
+      throw new NotFoundException(`User with ID ${email} not found`);
+    }
+    return user;
+  }
+
+  async updateUser(id: number, updateUserDto: UpdateUserDto): Promise<string> {
```

Comment:
```suggestion
  async update(id: number, data: Partial<User>): Promise<User> {
```
DTO has only relations with the controller, not inside repository layer